### DISCUSSION TITLE
hoa 3d decoder

### DISCUSSION THREAD

**Discussion by naokiapri** [grame-cncm/faustlibraries#144] [Q&A]
Hi! I'm trying to use 3d functions with hoa 3d encoder and cannot decode 3d soundfield, decoders form library do not have enough inputs. Am I missing something? there are 3d tools and no decoder for it?

**Comment by sletz** [comment]
Which functions are you using ? Hopefully @alainbonardi or @pierreguillot can help here.

**Reply by naokiapri** [reply]
ho.encoder3D and ho.optimInPhase3D. they are using full 3d b-format (4 outputs for 1st order) and ho.decoder is only 3 inputs at 1st order, looks like it is intended only for 2d b-format

**Comment by naokiapri** [comment]
solved by using decoder from ambitools package as component
http://www.sekisushai.net/ambitools

though it would be cool to find out if i'm missing something in hoa lib

**Reply by sletz** [reply]
Sure, hoping @alainbonardi or @pierreguillot can help here.

**Reply by alainbonardi** [reply]
Hi, sorry to answer you late. In the current state of hoa.lib, we just started to develop the 3D. Therefore there is no 3D decoder for the moment. Hopefully it will come by the end of the year. ho.decoder is a 2D decoder. Cheers.
